#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
from os.path import abspath, basename, dirname, splitext

PACKAGE = basename(dirname(dirname(abspath(__file__))))

gen = ParameterGenerator()

gen.add("error_propagation_count", int_t, 0, default=3, min=0,
        description="Number of tries until error is propagated to upper level")
gen.add("lookahead", double_t, 0, default=20.0, min=0.0, description="Distance to local goal")
gen.add("overlap", double_t, 0, default=1.0, min=0.0, description="Overlap between path segments")
gen.add("x_y_tolerance_increase", double_t, 0, default=1.0, min=0.0, max=10.0,
        description="Scaling of x and y tolerance")
gen.add("yaw_tolerance_increase", double_t, 0, default=1.0, min=0.0, max=10.0, description="Scaling of yaw tolerance")
gen.add("transformation_timeout", double_t, 0, default=0.1, min=0.0, max=10.0,
        description="Timeout for waiting for transforms")

exit(gen.generate(PACKAGE, PACKAGE + "_node", splitext(basename(__file__))[0]))
